spring-pulsar/src/main/java/org/springframework/pulsar/config/PulsarMessageRegistrar.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.config;
+
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.common.schema.SchemaType;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.annotation.MergedAnnotation;
+import org.springframework.pulsar.annotation.PulsarMessage;
+import org.springframework.pulsar.core.DefaultSchemaResolver;
+import org.springframework.pulsar.core.DefaultTopicResolver;
+import org.springframework.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Registers default schema and topic mappings based on {@code descriptors} property
+ *
+ * @author Aleksei Arsenev
+ */
+public class PulsarMessageRegistrar implements InitializingBean {
+
+	private List<MessageDescriptor> descriptors = new ArrayList<>();
+
+	@Autowired
+	private DefaultTopicResolver defaultTopicResolver = null;
+
+	@Autowired
+	private DefaultSchemaResolver defaultSchemaResolver = null;
+
+	public List<MessageDescriptor> getDescriptors() {
+		return descriptors;
+	}
+
+	public void setDescriptors(List<MessageDescriptor> descriptors) {
+		this.descriptors = descriptors;
+	}
+
+	@Override
+	public void afterPropertiesSet() throws Exception {
+		for (MessageDescriptor descriptor : this.descriptors) {
+			if (this.defaultTopicResolver != null && StringUtils.hasText(descriptor.topic)) {
+				this.defaultTopicResolver.addCustomTopicMapping(descriptor.message, descriptor.topic);
+			}
+
+			if (this.defaultSchemaResolver != null && descriptor.schemaType != SchemaType.NONE) {
+				Schema<Object> schema = this.defaultSchemaResolver
+						.resolveSchema(descriptor.schemaType, descriptor.message, descriptor.messageKeyType)
+						.orElseThrow();
+				this.defaultSchemaResolver.addCustomSchemaMapping(descriptor.message, schema);
+			}
+		}
+	}
+
+	public static final class MessageDescriptor {
```

Comment:
We have very similar structure in the SB side where we auto-configure in the [`spring.pulsar.defaults.type-mappings`](https://github.com/spring-projects/spring-boot/pull/34763/files#diff-bc2861ea587992c47c4fe7908a3f9de3f99fd88b35c5d68df39e086e524c383aR989-R1023). It may be a good idea if we move the above  structures from Spring Boot to here and replace the `MessageDescriptor` with them - that would keep the rules consistent. Wdyt? 

```java
	/**
	 * A mapping from message type to topic and/or schema info to use (at least one of
	 * {@code topicName} or {@code schemaInfo} must be specified.
	 *
	 * @param messageType the message type
	 * @param topicName the topic name
	 * @param schemaInfo the schema info
	 */
	public record TypeMapping(Class<?> messageType, String topicName, SchemaInfo schemaInfo) {
		public TypeMapping {
			Objects.requireNonNull(messageType, "messageType must not be null");
			if (topicName == null && schemaInfo == null) {
				throw new IllegalArgumentException("At least one of topicName or schemaInfo must not be null");
			}
		}
	}

	/**
	 * Represents a schema - holds enough information to construct an actual schema
	 * instance.
	 *
	 * @param schemaType schema type
	 * @param messageKeyType message key type (required for key value type)
	 */
	public record SchemaInfo(SchemaType schemaType, Class<?> messageKeyType) {
		public SchemaInfo {
			Objects.requireNonNull(schemaType, "schemaType must not be null");
			if (schemaType == SchemaType.NONE) {
				throw new IllegalArgumentException("schemaType NONE not supported");
			}
			if (schemaType != SchemaType.KEY_VALUE && messageKeyType != null) {
				throw new IllegalArgumentException("messageKeyType can only be set when schemaType is KEY_VALUE");
			}
		}
	}
```